Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,10 +98,26 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
-            guard
-                let templateDirectory = directories.first(where: { $0.basename == templateName })
-            else { throw InitServiceError.templateNotFound(templateName) }
-            let template = try templateLoader.loadTemplate(at: templateDirectory)
+            var template: Template
+            var tempPath: AbsolutePath?
+            if isURLFormattedString(templateName) {
+                let tempFolderName = AbsolutePath.current.removingLastComponent()
```

Comment:
FileHandler provides an [API](https://github.com/tuist/tuist/blob/main/Sources/TuistSupport/Utils/FileHandler.swift#L56) for creating a temporary directory and deleting it afterward automatically. I'd use that one instead and remove the logic down below that deletes the directories:

```swift
FileHandler.shared.inTemporaryDirectory { temporaryDirectory in
  // Do your stuff and you don't have to do any clean-up yourself.
}
```

Reply:
pretty weird that if I call it with the auto-deleting, the project is not created sucessfully, actually it said that 1 file is not found on folder. If I use `func inTemporaryDirectory(removeOnCompletion...`  with `removeOnCompletion` = `false` its works like a charm..